src/sprite_1.c

```diff
@@ -32,7 +32,7 @@ void sub_8002A3C(struct Unk_03002400 *r4)
     r4->unk8 = r6->unk0.unkC;
 
     if (!(r4->unk2E & 8)) {
-        gUnknown_030027A0[gUnknown_03002A84] = (struct Unk_03002EC0 *)r4;
+        gUnknown_030027A0[gUnknown_03002A84] = (struct BgHeader *)r4;
```

Comment:
This is why we can assume `Background` has a `->header` at `0x0` because it should produce the same code and no need to cast